app/controllers/events/participations_controller.rb

```diff
@@ -10,7 +10,9 @@ def create
   end
 
   def destroy
-    @event.participations.find_by(user_id: current_user.id).destroy
+    @event.with_lock do
+      @event.cancel_participation!(event: @event, user: current_user)
```

Comment:
```suggestion
      @event.cancel_participation!(user: current_user)
```

これで動くようにしましょう。

`@event`が実行しているのに`@event`を引数に渡す必要はないです。
cancel_participation!の中でselfを見れば`@event`ですので。

Reply:
なるほど🤔
`self`の使い方をよく理解していませんでした💦
インスタンス自身を表すキーワードが`self`なんですね。
修正しました🙇‍♂️